File: Exercism/fsharp/triangle/Triangle.fs
module Triangle

let countUniqueElems i = i |> Set.ofSeq |> Set.count

let isTriangleInequalityViolation t =
    let sorted = t |> Seq.sortDescending
    let largest = sorted |> Seq.head
    let remainder = sorted |> Seq.tail |> Seq.sum
    remainder < largest

let equilateral triangle =
    0. <> List.average triangle
    && 1 = countUniqueElems triangle

let isosceles triangle =
    (2 = countUniqueElems triangle
     || equilateral triangle)
    && not (isTriangleInequalityViolation triangle)

let scalene triangle =
    (3 = countUniqueElems triangle)
    && not (isTriangleInequalityViolation triangle)

File: Exercism/csharp/triangle/Triangle.cs
using System.Linq;

public static class Triangle
{
    private static int CountUniqueSides(params double[] triangle) => triangle.ToHashSet().Count();

    private static bool IsTriangleInequalityViolation(params double[] triangle) => triangle.OrderByDescending(i => i).Skip(1).Sum() < triangle.Max();

    public static bool IsScalene(params double[] triangle) => CountUniqueSides(triangle) == 3 && !IsTriangleInequalityViolation(triangle);

    public static bool IsIsosceles(params double[] triangle) => IsEquilateral(triangle) || CountUniqueSides(triangle) == 2 && !IsTriangleInequalityViolation(triangle);

    public static bool IsEquilateral(params double[] triangle) => CountUniqueSides(triangle) == 1 && (triangle.Average() != 0);
}